crates/bevy_ecs/src/world/mod.rs

```diff
@@ -2796,4 +2987,63 @@ mod tests {
         let mut world = World::new();
         world.spawn(());
     }
+
+    #[test]
+    fn iterate_resources() {
```

Comment:
These need a test case to ensure that iteration doesn't panic if you remove a resource. Also for miri.